src/main/java/org/terasology/gooeyDefence/components/enemies/EntrancePathComponent.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.gooeyDefence.components.enemies;
+
+import org.terasology.gooeyDefence.PathfindingSystem;
+import org.terasology.math.geom.Vector3i;
+
+import java.util.List;
+
+/**
+ * Moves the enemy along a path from an entrance to the shrine.
+ * <p>
+ * Doesn't store the path internally to reduce on memory, instead stores a
+ * reference to the PathfindingSystem that holds the path.
+ *
+ * @see PathfindingSystem
+ */
+public class EntrancePathComponent implements PathComponent {
+    private int step;
+    private int entranceId;
+    private Vector3i goal;
+    private PathfindingSystem pathManager;
+
+    /**
+     * Empty constructor for deserialisation.
+     */
+    private EntrancePathComponent() {
+    }
+
+    /**
+     * Create a new entrance path component specifying the position along the path to start at.
+     *
+     * @param entranceId  The ID of the entrance
+     * @param pathManager The PathfindingSystem the path is stored in
+     * @param startStep   The step to start from.
+     */
+    public EntrancePathComponent(int entranceId, PathfindingSystem pathManager, int startStep) {
+        this.entranceId = entranceId;
+        this.pathManager = pathManager;
+        step = Math.max(Math.min(pathManager.getPath(entranceId).size() - 1, startStep), 0);
```

Comment:
I assume you're doing the max because startStep may be a negative value (because the constructor below without the startStep doesn't have max()) which was not clear to me immediately. So i think it can be clarified (maybe do Math.max(startStep, 0) first?). Also if a negative value to startStep is invalid should an exception should be thrown?